src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -60,15 +64,26 @@ object RockSlicer {
       !block.faces.exists(_.processorJoint)
     }
 
-    // Process processor blocks before continueing with remaining blocks
-    // Collect all blocks that contain processor joints from all nodes
-    val allProcessorBlocks = processorBlocks.collect()
+    // Process processor blocks before continuing with remaining blocks
     // Search blocks for matching processor joints
-    if (allProcessorBlocks.length > 0) {
-      val reconstructedBlocks = mergeBlocks(allProcessorBlocks, Seq.empty[Block], globalOrigin)
+    if (processorBlocks.count() > 0) {
+      val treeReduceBlockPairsRDD = processorBlocks.map{ blocks => (Seq(blocks), Seq.empty[Block])}
```

Comment:
The tree reduce stuff generally looks good. Could you add some comments to explain a bit about what's going on at each step, e.g. mentioning what it means to be an orphan block?